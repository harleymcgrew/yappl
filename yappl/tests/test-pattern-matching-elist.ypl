fun bool:partytime int[]:a =  
    match a with 
     b::c -> ((~print_line b) && (~partytime c ))
   | [] -> false
in 
let int[]:x = [5,4,3,1,0] in
~print_line ( ~partytime x );
match x with
   _ :: n -> ~print_line n
 | _ -> ~print_line 0
