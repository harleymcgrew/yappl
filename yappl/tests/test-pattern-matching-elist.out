5
4
3
1
0
false
[4,3,1,0]
